src/api/app/controllers/webui/requests_listing_controller.rb

```diff
@@ -1,5 +1,7 @@
 class Webui::RequestsListingController < Webui::WebuiController
-  before_action :require_login
+  before_action :require_login, unless: -> { params[:project].present? }
+  before_action :check_bata_user, if: -> { params[:project].present? }
```

Comment:
Why all those `if project`?

Reply:
For projects `require_login` is not required. `check_bata_user` is only for projects because this PR is only looking into project/requests

Reply:
I think this controller is trying to do three very similar, but not quite the same, things. It's getting more complex. What about extracting the common bits into a Concern and write three controllers?

Reply:
It is like going one step back. We had three different controllers and moved to this new controller so we can avoid code repetition. All three use same view/filters etc.

Reply:
All three use almost the same but not quite, if not, you would not need the unless/if filters on the before_actions...

Reply:
project/package are using same filters. user/requests are a bit different. We would need a refactoring with performance improvement. I can't have it in this pr so it is not blocked

Reply:
Fair enough